### PR TITLE
Adding parsing of headers and cookies

### PR DESCRIPTION
**Pull request by tpavs**

### CODE REVIEW

File: authentication/authentication.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	UserIdCookie = "cuid"
+	UserTokenCookie = "cutk"
+)
+
+type Authenticator interface {
+    Authorize() bool
+}
+
+func getHeaderOrCookie(hKey string, cKey string, r *http.Request) string {
+	hVal := r.Header.Get(hKey)
```
Comment (lickel) [inline authentication/authentication.go]: Is this case insensitive?

File: authentication/authentication.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	return ""
+}
+
+type TokenPair struct {
+	userId string
+	token string
+}
+
+func getTokenAuthentication(r *http.Request, deviceId string) (*Authenticator, func(resp http.ResponseWriter) ) {
+	authorizationString := r.Header.Get(AuthorizationHeader)
```
Comment (lickel) [inline authentication/authentication.go]: Should probably get lower-cased since this was also supposed to be case insensitive

Comment (tpavs) [reply authentication/authentication.go]: Library is case insensitive 👍

File: authentication/authentication.go (hunk 1 above)
Comment (tpavs) [reply authentication/authentication.go]: Yes
https://golang.org/pkg/net/http/#Header.Get

File: authentication/authentication.go (hunk 2 above)
Comment (lickel) [reply authentication/authentication.go]: Sorry, I actually meant `authorizationString` 🙂

File: authentication/authentication.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	hVal := r.Header.Get(hKey)
+	cVal, _ := r.Cookie(cKey)
+
+	if hVal == "" {
+		if cVal != nil {
+			return cVal.Value
+		}
+	}
+
+	return ""
```
Comment (proflayton) [inline authentication/authentication.go]: shouldn't this be `return hVal`?

File: authentication/authentication.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		tokenPair = &TokenPair{userIdCookie.Value, tokenCookie.Value}
+	}
+
+	if tokenPair == nil {
+		return nil, func(resp http.ResponseWriter) {
+			http.Error(resp, "Malformed crew token", http.StatusBadRequest)
+		}
+	}
+
+	var result Authenticator = CrewIdTokenAuthenticator{
```
Comment (proflayton) [inline authentication/authentication.go]: little confused here. This should be a Syntax error, did you accidentally put in `Authenticator`?

**Pull request merged**